api/src/main/java/org/openmrs/api/db/hibernate/HibernateContextDAO.java

```diff
@@ -93,126 +93,135 @@ public void setUserDAO(UserDAO userDao) {
-				} else {
-					candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP, String.valueOf(System
-						.currentTimeMillis()));
-					throw new ContextAuthenticationException(
-						"Invalid number of connection attempts. Please try again later.");
 				}
+			} catch (NumberFormatException e) {
+				log.warn("bad value stored in {} user property: {}", OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP, lockoutTimeString);
 			}
+		}
+		return false;
+	}
 
-			Object[] passwordAndSalt = (Object[]) session
-				.createNativeQuery("select password, salt from users where user_id = ?1")
-				.addScalar("password", StandardBasicTypes.STRING).addScalar("salt", StandardBasicTypes.STRING)
-				.setParameter(1, candidateUser.getUserId()).uniqueResult();
+	private boolean isPasswordValid(Session session, User candidateUser, String password) {
+		Object[] passwordAndSalt = (Object[]) session
+			.createNativeQuery("select password, salt from users where user_id = ?1")
+			.addScalar("password", StandardBasicTypes.STRING)
+			.addScalar("salt", StandardBasicTypes.STRING)
+			.setParameter(1, candidateUser.getUserId())
+			.uniqueResult();
 
-			String passwordOnRecord = (String) passwordAndSalt[0];
-			String saltOnRecord = (String) passwordAndSalt[1];
+		String passwordOnRecord = (String) passwordAndSalt[0];
+		String saltOnRecord = (String) passwordAndSalt[1];
 
-			// if the username and password match, hydrate the user and return it
-			if (passwordOnRecord != null && Security.hashMatches(passwordOnRecord, password + saltOnRecord)) {
-				// hydrate the user object
-				candidateUser.getAllRoles().size();
-				candidateUser.getUserProperties().size();
-				candidateUser.getPrivileges().size();
+		return passwordOnRecord != null && Security.hashMatches(passwordOnRecord, password + saltOnRecord);
+	}
 
-				// only clean up if the were some login failures, otherwise all should be clean
-				int attempts = getUsersLoginAttempts(candidateUser);
-				if (attempts > 0) {
-					candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOGIN_ATTEMPTS, "0");
-					candidateUser.removeUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP);
-					saveUserProperties(candidateUser);
-				}
+	private void resetFailedLoginAttempts(User candidateUser) {
+		int attempts = getUsersLoginAttempts(candidateUser);
+		if (attempts > 0) {
+			candidateUser.setUserProperty(OpenmrsConstants.USER_PROPERTY_LOGIN_ATTEMPTS, OpenmrsConstants.ZERO_LOGIN_ATTEMPTS_VALUE);
+			candidateUser.removeUserProperty(OpenmrsConstants.USER_PROPERTY_LOCKOUT_TIMESTAMP);
+		}
+	}
+	
+	private void setLastLoginTime(User candidateUser) {
+		candidateUser.setUserProperty(
+			OpenmrsConstants.USER_PROPERTY_LAST_LOGIN_TIMESTAMP,
+			String.valueOf(System.currentTimeMillis())
+		);
+	}
 
-				// skip out of the method early (instead of throwing the exception)
-				// to indicate that this is the valid user
-				return candidateUser;
-			} else {
```

Comment:
The changes in this file are way much more than i would expect. Can we limit them to just the ticket requirements?

Reply:
Reverted